cdap-docs/developers-manual/source/building-blocks/datasets/fileset.rst

```diff
@@ -130,3 +129,211 @@ See the Apache™ Twill®
+``TextOutputFormat.``, and for the output format, we specify that the separator between fields is a comma.
+The difference to a FileSet is that this Dataset is partitioned by league and season. This means that every file
+added to this Dataset must have a partitioning key with a unique combination of league and season.
+
+Reading and Writing PartitionedFileSets
+=======================================
+
+You can interact with the files in a PartitionedFileSet directly, through the ``Location`` abstraction
+of the file system. This is similar to a FileSet, but instead of a relative path, you specify a
+partition key to obtain a Partition; you can then get a Location from that Partition.
+
+For example, to read the content of a partition::
+
+      PartitionKey key = PartitionKey.builder().addStringField(...)
+                                               .addIntField(...)
+                                               .build());
+      Partition partition = dataset.getPartition(key);
+      if (partition != null) {
+        try {
+          Location location = partition.getLocation();
+          InputStream inputStream = location.getInputStream();
+          ...
+        } catch (IOException e) {
+          ...
+        }
+      }
+
+Note that if the partition was written with MapReduce, the location is actually a directory
+that contains part files. In that case, list the files in the directory to find the part files::
+
+    for (Location file : location.list()) {
+      if (file.getName().startsWith("part")) {
+        InputStream inputStream = location.getInputStream();
+        ...
+      }
+    }
+
+Instead of reading a single partition, you can also specify a PartitionFilter to query the
+partitioned file set for all partitions whose keys match that filter. The PartitionFilter
+can specify either a exact value (en equality condition) or a range for the value of each
+field in the dataset's partitioning. For example, the following code reads all partitions
+for the NFL and the eighties' seasons::
+
+      PartitionFilter filter = PartitionFilter.builder().addValueCondition("league", "nfl")
+                                                        .addRangeCondition("season", 1980, 1990)
+                                                        .build());
+      Set<Partition> partitions = dataset.getPartitions(filter);
+      for (partition : partitions) {
+        try {
+          Location location = partition.getLocation();
+          InputStream inputStream = location.getInputStream();
+          ...
+        } catch (IOException e) {
+          ...
+        }
+      }
+
+Note that the upper bound for the season (1990) is exclusive, that is, the 1990 season is not
+included in returned partitions. For a range condition, either the lower or the upper bound may
+be null, meaning that the filter in unbounded in that direction.
+
+Writing a partition is similar, but instead of a Partition, you get a ``PartitionOutput``
```

Comment:
actually, "writing to" would suggest that the partition already exists. But here we are writing a new partition. I think it is better to say, "Adding a partition is similar".